base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs

```diff
@@ -393,13 +398,14 @@ impl UnconfirmedPool {
                 {
                     update_candidate_transactions_to_select.remove(&tx_key);
                     *update_total_transaction_weight -= prioritized_transaction.weight;
-                    *update_total_transaction_fees -= prioritized_transaction.transaction.body.get_total_fee().0;
+                    *update_total_transaction_fees -= prioritized_transaction.transaction.body.get_total_fee()?.0;
```

Comment:
Needs an underflow check or a comment saying why it can't underflow. (lines 400 and 401)